### DISCUSSION TITLE
Best way to drop an outbound email

### DISCUSSION THREAD

**Discussion by lnedry** [haraka/Haraka#3267] [Q&A] [answered]
What is the best way to drop an outbound email?

Once in a while a web hosted customer will configure a contact us form that sends an email with a From domain that isn't hosted here. I tried DENYDISCONNECT in the hook_data_post but the email reject message gets sent to the From address.

**Comment by msimerson** [comment, marked as answer]
- A1: Off the top of my head, create a plugin that listens on `queue` and `queue_outbound` and returns `OK` for messages that match your criteria.
- A2: In haraka/Haraka#3265, I have implemented a check that requires the Envelope Sender domain to match the domain used during SMTP AUTH. Use a test like that to prevent emails from non-local domains being accepted in the first place.
